golangci-lint errors

```
Linting go files...
pkg/pipelinerunmetrics/injection.go:81:35: G118: context cancellation function returned by WithCancel/WithTimeout/WithDeadline is not called (gosec)
        ctx, cancel := context.WithCancel(ri.ctx)
                                         ^
pkg/adapter/incoming_test.go:836:30: net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext (noctx)
                        req := httptest.NewRequest(tt.args.method,
                                                  ^
pkg/adapter/incoming_test.go:1124:28: net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext (noctx)
                        req: httptest.NewRequest(http.MethodPost,
                                                ^
pkg/adapter/incoming_test.go:1140:29: net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext (noctx)
                                r := httptest.NewRequest(http.MethodPost,
                                                        ^
pkg/adapter/incoming_test.go:1222:28: net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext (noctx)
        req := httptest.NewRequest(http.MethodPost,
                                  ^
pkg/adapter/incoming_test.go:1326:30: net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext (noctx)
                        req := httptest.NewRequest(tt.method, tt.url, strings.NewReader(tt.body))
                                                  ^
pkg/provider/github/app/token_test.go:147:45: net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext (noctx)
                        ip := NewInstallation(httptest.NewRequest(http.MethodGet, "http://localhost", strings.NewReader("")), run, &v1alpha1.Repository{}, &github.Provider{}, tt.namespace.GetName())
                                                                 ^
pkg/provider/github/app/token_test.go:209:43: net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext (noctx)
        ip := NewInstallation(httptest.NewRequest(http.MethodGet, "http://localhost", strings.NewReader("")), run, &v1alpha1.Repository{}, &github.Provider{}, testNamespace.GetName())
                                                 ^
pkg/provider/github/app/token_test.go:212:28: net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext (noctx)
        req := httptest.NewRequest(http.MethodGet, "http://localhost", strings.NewReader(""))
                                  ^
pkg/provider/github/app/token_test.go:396:45: net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext (noctx)
                        ip := NewInstallation(httptest.NewRequest(http.MethodGet, "http://localhost", strings.NewReader("")), run, &v1alpha1.Repository{}, &github.Provider{}, testNamespace.GetName())
                                                                 ^
pkg/provider/github/app/token_test.go:415:44: net/http/httptest.NewRequest must not be called. use net/http/httptest.NewRequestWithContext (noctx)
                        ip = NewInstallation(httptest.NewRequest(http.MethodGet, "http://localhost", strings.NewReader("")), run, repo, gprovider, testNamespace.GetName())
                                                                ^
pkg/provider/github/github.go:900:2: Consider preallocating baseFields with capacity 18 + len(kv) (prealloc)
        baseFields := []any{
        ^
12 issues:
* gosec: 1
* noctx: 10
* prealloc: 1
make: *** [Makefile:98: lint-go] Error 1
```

--- a/pkg/adapter/incoming_test.go
+++ b/pkg/adapter/incoming_test.go
@@ -1,6 +1,7 @@
 package adapter
 
 import (
+	"context"
 	"fmt"
 	"net/http"
 	"net/http/httptest"
@@ -833,7 +834,7 @@ func Test_listener_detectIncoming(t *testing.T) {
 			}
 
 			// make a new request
-			req := httptest.NewRequest(tt.args.method,
+			req := httptest.NewRequestWithContext(ctx, tt.args.method,
 				fmt.Sprintf("http://localhost%s?repository=%s&secret=%s&pipelinerun=%s&branch=%s&namespace=%s", tt.args.queryURL,
 					tt.args.queryRepository, tt.args.querySecret, tt.args.queryPipelineRun, tt.args.queryBranch, tt.args.queryNamespace),
 				strings.NewReader(tt.args.incomingBody))
@@ -1121,7 +1122,7 @@ func Test_detectIncoming_legacy_warning(t *testing.T) {
 	}{
 		{
 			name: "legacy mode - params in URL",
-			req: httptest.NewRequest(http.MethodPost,
+			req: httptest.NewRequestWithContext(ctx, http.MethodPost,
 				"http://localhost/incoming?repository=test-good&secret=verysecrete&pipelinerun=pipelinerun1&branch=main",
 				strings.NewReader("")),
 			body:          nil,
@@ -1137,7 +1138,7 @@ func Test_detectIncoming_legacy_warning(t *testing.T) {
 					"secret": "verysecrete",
 					"params": {"foo": "bar"}
 				}`
-				r := httptest.NewRequest(http.MethodPost,
+				r := httptest.NewRequestWithContext(ctx, http.MethodPost,
 					"http://localhost/incoming",
 					strings.NewReader(payload))
 				r.Header.Set("Content-Type", "application/json")
@@ -1219,7 +1220,7 @@ func Test_detectIncoming_body_params_are_parsed(t *testing.T) {
 		"secret": "verysecrete",
 		"params": {"foo": "bar", "bar": "baz"}
 	}`
-	req := httptest.NewRequest(http.MethodPost,
+	req := httptest.NewRequestWithContext(ctx, http.MethodPost,
 		"http://localhost/incoming",
 		strings.NewReader(payload))
 	req.Header.Set("Content-Type", "application/json")
@@ -1323,7 +1324,7 @@ func Test_parseIncomingPayload(t *testing.T) {
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			req := httptest.NewRequest(tt.method, tt.url, strings.NewReader(tt.body))
+			req := httptest.NewRequestWithContext(context.Background(), tt.method, tt.url, strings.NewReader(tt.body))
 			if tt.headers != nil {
 				req.Header = tt.headers
 			}

--- a/pkg/pipelinerunmetrics/injection.go
+++ b/pkg/pipelinerunmetrics/injection.go
@@ -78,7 +78,7 @@ var _ controller.Informer = (*recorderInformer)(nil)
 // Run starts the recorder informer in a goroutine.
 func (ri *recorderInformer) Run(stopCh <-chan struct{}) {
 	// Turn the stopCh into a context for reporting metrics.
-	ctx, cancel := context.WithCancel(ri.ctx)
+	ctx, cancel := context.WithCancel(ri.ctx) // #nosec G118 -- cancel is called in the goroutine below
 	go func() {
 		<-stopCh
 		cancel()

--- a/pkg/provider/github/app/token_test.go
+++ b/pkg/provider/github/app/token_test.go
@@ -144,7 +144,7 @@ func Test_GenerateJWT(t *testing.T) {
 				},
 			}
 
-			ip := NewInstallation(httptest.NewRequest(http.MethodGet, "http://localhost", strings.NewReader("")), run, &v1alpha1.Repository{}, &github.Provider{}, tt.namespace.GetName())
+			ip := NewInstallation(httptest.NewRequestWithContext(ctx, http.MethodGet, "http://localhost", strings.NewReader("")), run, &v1alpha1.Repository{}, &github.Provider{}, tt.namespace.GetName())
 			token, err := ip.GenerateJWT(ctx)
 			if tt.wantErr {
 				assert.Assert(t, err != nil)
@@ -206,10 +206,10 @@ func Test_GetAndUpdateInstallationID(t *testing.T) {
 	ctx = info.StoreCurrentControllerName(ctx, "default")
 	ctx = info.StoreNS(ctx, testNamespace.GetName())
 
-	ip := NewInstallation(httptest.NewRequest(http.MethodGet, "http://localhost", strings.NewReader("")), run, &v1alpha1.Repository{}, &github.Provider{}, testNamespace.GetName())
+	ip := NewInstallation(httptest.NewRequestWithContext(ctx, http.MethodGet, "http://localhost", strings.NewReader("")), run, &v1alpha1.Repository{}, &github.Provider{}, testNamespace.GetName())
 	jwtToken, err := ip.GenerateJWT(ctx)
 	assert.NilError(t, err)
-	req := httptest.NewRequest(http.MethodGet, "http://localhost", strings.NewReader(""))
+	req := httptest.NewRequestWithContext(ctx, http.MethodGet, "http://localhost", strings.NewReader(""))
 	repo := &v1alpha1.Repository{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: "repo",
@@ -393,7 +393,7 @@ func TestGetAndUpdateInstallationID_Fallbacks(t *testing.T) {
 			ctx = info.StoreCurrentControllerName(ctx, "default")
 			ctx = info.StoreNS(ctx, testNamespace.GetName())
 
-			ip := NewInstallation(httptest.NewRequest(http.MethodGet, "http://localhost", strings.NewReader("")), run, &v1alpha1.Repository{}, &github.Provider{}, testNamespace.GetName())
+			ip := NewInstallation(httptest.NewRequestWithContext(ctx, http.MethodGet, "http://localhost", strings.NewReader("")), run, &v1alpha1.Repository{}, &github.Provider{}, testNamespace.GetName())
 			jwtToken, err := ip.GenerateJWT(ctx)
 			assert.NilError(t, err)
 
@@ -412,7 +412,7 @@ func TestGetAndUpdateInstallationID_Fallbacks(t *testing.T) {
 			gprovider.SetGithubClient(fakeghclient)
 			t.Setenv("PAC_GIT_PROVIDER_TOKEN_APIURL", serverURL)
 
-			ip = NewInstallation(httptest.NewRequest(http.MethodGet, "http://localhost", strings.NewReader("")), run, repo, gprovider, testNamespace.GetName())
+			ip = NewInstallation(httptest.NewRequestWithContext(ctx, http.MethodGet, "http://localhost", strings.NewReader("")), run, repo, gprovider, testNamespace.GetName())
 			_, token, installationID, err := ip.GetAndUpdateInstallationID(ctx)
 
 			if tt.wantErr {

--- a/pkg/provider/github/github.go
+++ b/pkg/provider/github/github.go
@@ -864,7 +864,8 @@ func (v *Provider) debugCommentPhase(event *info.Event, trace commentTraceLogCon
 		pr = event.PullRequestNumber
 	}
 
-	baseFields := []any{
+	baseFields := make([]any, 0, 18+len(kv))
+	baseFields = append(baseFields,
 		"phase", phase,
 		"organization", org,
 		"repository", repo,
@@ -874,7 +875,7 @@ func (v *Provider) debugCommentPhase(event *info.Event, trace commentTraceLogCon
 		"marker_hash", trace.markerHash,
 		"marker_len", trace.markerLen,
 		"controller_label", trace.controllerLabel,
-	}
+	)
 	v.Logger.Debugw("github comment dedup flow", append(baseFields, kv...)...)
 }
 

